web/models.py

```
@@ -213,6 +215,25 @@ def save(self, *args, **kwargs):
+                return False
+        from django.utils import timezone
+
+        self.facts.append({"text": fact_text, "generated_at": timezone.now().isoformat()})
+        self.save(update_fields=["facts"])
+        return True
+
+    def get_random_fact(self) -> Optional[str]:
+        import random
+
+        if self.facts:
+            return random.choice(self.facts)["text"]
```

Ruff (0.8.2)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)